arch/arm/soc/atmel_sam/same70/soc.c

```diff
@@ -227,6 +227,9 @@ static int atmel_same70_init(struct device *arg)
 
 	key = irq_lock();
 
+	SCB_EnableICache();
```

Comment:
One question: worth moving this to core ARM code instead of at the SOC level?

Reply:
Other SoCs are already managing this themselves, so I just guessed (also because this is where @aurel32 suggested it go) that there are vendor-specific constraints with ordering or whatnot.

But looking at the code, it's generic ARM CMSIS stuff and doesn't appear to be vendor-related.  So I'd guess yeah?  But I'm a total ARM noob so was afraid to make that call.

Reply:
We'd have to make sure the non i/d cache soc's do the right thing.  I presume this is a empty macro on ARM SoC's which don't support cache.

Reply:
This is actually a function and not a macro, defined in `ext/hal/cmsis/Include/core_cm7.h`, so it is only available on Cortex-M7 based SoCs.

Reply:
I think the best possible solution is to keep it at soc level. And let the Cortex-M7 based SOCs enable it in soc.c.

Reply:
Agree with keeping the instruction cache enabling on SOC level. That's what we do for Nordic SOCs, AFAICT:

```
#ifdef CONFIG_NRF_ENABLE_ICACHE
	/* Enable the instruction cache */
	NRF_NVMC->ICACHECNF = NVMC_ICACHECNF_CACHEEN_Msk;
#endif
```